service/src/main/java/uk/gov/hmcts/reform/fpl/service/sdo/ListGatekeepingHearingDecider.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.fpl.service.sdo;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.enums.DirectionsOrderType;
+import uk.gov.hmcts.reform.fpl.events.GatekeepingOrderEvent;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.StandardDirectionOrder;
+
+import java.util.Optional;
+
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionsOrderType.SDO;
+import static uk.gov.hmcts.reform.fpl.enums.DirectionsOrderType.UDO;
+import static uk.gov.hmcts.reform.fpl.enums.notification.GatekeepingOrderNotificationGroup.SDO_OR_UDO_AND_NOP;
+
+@Component
+public class ListGatekeepingHearingDecider {
+
+    public Optional<GatekeepingOrderEvent> buildEventToPublish(final CaseData caseData) {
+
+        final DirectionsOrderType directionsOrderType;
+        final StandardDirectionOrder standardDirectionOrder;
```

Comment:
could we maybe rename this to something that fits both orders. I know they're the same type, but the name sort of implies the standard rather than urgent